composer/profiler/torch_profiler.py

```diff
@@ -54,84 +53,242 @@ class TorchProfiler(Callback):
 
     Args:
-        tensorboard_trace_handler_dir (str): Directory to store trace results.
-            Relative to the run_directory. Defaults to ``torch_profiler`` in the
-            run directory.
-        tensorboard_use_gzip (bool, optional):
-            Whether to use gzip for the trace. Defaults to False.
-        record_shapes (bool, optional): Whether to record tensor shapes.
-            Defaults to False.
-        profile_memory (bool, optional): Whether to profile memory.
-            Defaults to True.
-        with_stack (bool, optional): Whether to record stack info.
-            Defaults to False.
-        with_flops (bool, optional): Whether to estimate flops for operators.
-            Defaults to True.
+        folder (str, optional): Format string for the folder containing the Torch Profiler trace files.
+            Defaults to ``'{run_name}/torch_traces'``.
+
+            The following format variables are available:
+
+            +------------------------+-------------------------------------------------------+
+            | Variable               | Description                                           |
+            +========================+=======================================================+
+            | ``{run_name}``         | The name of the training run. See                     |
+            |                        | :attr:`.Logger.run_name`.                             |
+            +------------------------+-------------------------------------------------------+
+            | ``{rank}``             | The global rank, as returned by                       |
+            |                        | :func:`~composer.utils.dist.get_global_rank`.         |
+            +------------------------+-------------------------------------------------------+
+            | ``{local_rank}``       | The local rank of the process, as returned by         |
+            |                        | :func:`~composer.utils.dist.get_local_rank`.          |
+            +------------------------+-------------------------------------------------------+
+            | ``{world_size}``       | The world size, as returned by                        |
+            |                        | :func:`~composer.utils.dist.get_world_size`.          |
+            +------------------------+-------------------------------------------------------+
+            | ``{local_world_size}`` | The local world size, as returned by                  |
+            |                        | :func:`~composer.utils.dist.get_local_world_size`.    |
+            +------------------------+-------------------------------------------------------+
+            | ``{node_rank}``        | The node rank, as returned by                         |
+            |                        | :func:`~composer.utils.dist.get_node_rank`.           |
+            +------------------------+-------------------------------------------------------+
```

Comment:
[nit] Could we use an f-string or substitution here to avoid copying this table from `composer.utils.file_helpers`?